src/soc/ibm/power9/istep_18_12.c

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <console/console.h>
+#include <cpu/power/istep_18.h>
+#include <timer.h>
+
+static void init_tod_node(void)
+{
+
+    write_scom(PERV_TOD_TX_TTYPE_2_REG, PPC_BIT(PERV_TOD_TX_TTYPE_2_REG_TRIGGER));
+    write_scom(PERV_TOD_LOAD_TOD_MOD_REG, PPC_BIT(PERV_TOD_LOAD_TOD_MOD_REG_FSM_TRIGGER));
+    write_scom(PERV_TOD_TX_TTYPE_5_REG, PPC_BIT(PERV_TOD_TX_TTYPE_5_REG_TRIGGER));
+    write_scom(PERV_TOD_LOAD_TOD_REG, PPC_BITMASK(50, 61));
+    write_scom(PERV_TOD_START_TOD_REG, PPC_BIT(PERV_TOD_START_TOD_REG_FSM_TRIGGER));
+    write_scom(PERV_TOD_TX_TTYPE_4_REG, PPC_BIT(PERV_TOD_TX_TTYPE_4_REG_TRIGGER));
+
+    // wait until TOD is running
+    wait_us(1000, read_scom(PERV_TOD_FSM_REG) & PPC_BIT(PERV_TOD_FSM_REG_IS_RUNNING));
```

Comment:
Please add check for timeout.

Reply:
Added a check for timeout with an error message